From 69593c76877f9f8e6455c178f0091a8b6a5335cf Mon Sep 17 00:00:00 2001
From: Vagrant Cascadian <vagrant@debian.org>
Date: Tue, 5 Jun 2018 13:08:46 -0700
Subject: [PATCH 3/4] arm: odroid: Enable distro_bootcmd support.

Enable distro_bootcmd for a standardized boot process across multiple
platforms.

Series-to: u-boot
---
 include/configs/odroid.h | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/include/configs/odroid.h b/include/configs/odroid.h
index c795ee8c74..e6224ba2f1 100644
--- a/include/configs/odroid.h
+++ b/include/configs/odroid.h
@@ -41,7 +41,7 @@
 
 /* Console configuration */
 
-#define CONFIG_BOOTCOMMAND		"run autoboot"
+#define CONFIG_BOOTCOMMAND		"run distro_bootcmd ; run autoboot"
 #define CONFIG_DEFAULT_CONSOLE		"ttySAC1,115200n8"
 
 #define CONFIG_SYS_INIT_SP_ADDR	(CONFIG_SYS_LOAD_ADDR \
@@ -83,6 +83,12 @@
 	"bl2 raw 0x1f 0x1d;" \
 	"tzsw raw 0x83f 0x138\0"
 
+#define BOOT_TARGET_DEVICES(func) \
+	func(MMC, mmc, 1) \
+	func(MMC, mmc, 0)
+
+#include <config_distro_bootcmd.h>
+
 /*
  * Bootable media layout:
  * dev:    SD   eMMC(part boot)
@@ -167,7 +173,8 @@
 	"ramdisk_addr_r=0x42000000\0" \
 	"scriptaddr=0x42000000\0" \
 	"fdt_addr_r=0x40800000\0" \
-	"kernel_addr_r=0x41000000\0"
+	"kernel_addr_r=0x41000000\0" \
+	BOOTENV
 
 /* GPT */
 
-- 
2.11.0

